### PR TITLE
langgraph[patch]: rename state_modifier -> preprocess_model_inputs in create_react_agent

### PR DESCRIPTION
**Pull request by baskaryan**

### CODE REVIEW

**Comment by vbarda** [comment]
@baskaryan i think we should just make it work with `state_modifier` instead of adding another kwarg

**Comment by baskaryan** [comment]
> @baskaryan i think we should just make it work with `state_modifier` instead of adding another kwarg

sticking to state_modifier makes our code cleaner but think it makes the user experience more confusing. think if prompt was available ppl wouldn't use state_modifier. and makes the transition from langchain agent executor much simpler

**Comment by vbarda** [comment]
> > @baskaryan i think we should just make it work with `state_modifier` instead of adding another kwarg
> 
> sticking to state_modifier makes our code cleaner but think it makes the user experience more confusing. think if prompt was available ppl wouldn't use state_modifier. and makes the transition from langchain agent executor much simpler

updated this to use a new name - `preprocess_model_inputs`

File: libs/langgraph/langgraph/prebuilt/chat_agent_executor.py [hunk 1]
```diff
@@ -49,89 +49,110 @@ class AgentState(TypedDict):
...
     raise ValueError(
         f"Got unexpected type for `messages_modifier`: {type(messages_modifier)}"
     )
 
 
-def _get_model_preprocessing_runnable(
-    state_modifier: Optional[StateModifier],
+def _get_model_input_preprocessor(
+    preprocess_model_inputs: Optional[ModelInputPreprocessor],
+    state_modifier: Optional[ModelInputPreprocessor],
```
Comment (baskaryan) [inline libs/langgraph/langgraph/prebuilt/chat_agent_executor.py:115]: why do we need both (for a private method)

Comment (vbarda) [reply libs/langgraph/langgraph/prebuilt/chat_agent_executor.py:115]: we don't necessarily, but putting the logic for picking the preprocessor into a single private function keeps the actual `create_react_agent` a bit cleaner imo

**Pull request closed without merging**